plugins/catalog-backend/src/ingestion/processors/AnnotateLocationEntityProcessor.test.ts

```diff
@@ -52,8 +57,110 @@ describe('AnnotateLocationEntityProcessor', () => {
         metadata: {
           name: 'my-component',
           annotations: {
-            'backstage.io/managed-by-location': 'url:my-location',
-            'backstage.io/managed-by-origin-location': 'url:my-origin-location',
+            'backstage.io/managed-by-location':
+              'url:https://github.com/backstage/backstage/blob/master/packages/app/catalog-info.yaml',
+            'backstage.io/managed-by-origin-location':
+              'url:https://github.com/backstage/backstage/blob/master/catalog-info.yaml',
+            'backstage.io/view-url':
+              'https://github.com/backstage/backstage/blob/master/packages/app/catalog-info.yaml',
+            'backstage.io/edit-url':
+              'https://github.com/backstage/backstage/edit/master/packages/app/catalog-info.yaml',
+            'backstage.io/source-location':
+              'https://github.com/backstage/backstage/tree/master/packages/app/',
+          },
+        },
+      });
+    });
+
+    it('does not override existing annotations', async () => {
+      const entity: Entity = {
+        apiVersion: 'backstage.io/v1alpha1',
+        kind: 'Component',
+        metadata: {
+          name: 'my-component',
+          annotations: {
+            'backstage.io/view-url': 'https://example.com/view',
+            'backstage.io/edit-url': 'https://example.com/edit',
+            'backstage.io/source-location': 'https://example.com/source',
```

Comment:
`backstage.io/source-location` uses the catalog location format, i.e. `url: https://example.com/source`, noticed that's not assumed throughout this PR 😅 

Kinda wanna challenge that a lil bit though @freben, what kind of non-url location would actually be useful? I'm assuming stuff like GitHub orgs would not have a `source-location` as they're not source code. For org-specific custom locations we can't handle those here anyway, so is there actually any value to including the location type? With that I'm ofc also arguing that it would be `source-url` :p

Reply:
I was thinking things like `sysmodel:<id>`. But besides, whether it happens to be a url normally, it is location like, isn't it? And other locations have a type and target. I was intending for this field to be consumable by processors / general code. But you mean that it'd be descoped to be a browser-only thing for human consumption then right?

Reply:
Doesn't it make a lot of sense to have the ability for tooling to e.g. traverse to the pointrd-at repo and list the number of unmerged pull requests etc?

Bear in mind - this annotation is essentially the _replacement_ for managed-by-location when what you point to isn't where the actual yaml file is

Reply:
It wouldn't be something like `sysmodel:<id>` tho right? that's be the manage-by-location. `source-location` is strictly the location of the source code, and doesn't really have that much to do with locations right?

Reply:
I meant that managed-by-location is strictly "where this yaml was sourced from" and source-location is "what i am describing" which can be two different things

Reply:
In the sysmodel example they could for example be `sysmodel:*` and sysmodel:<id>`, respectively

Reply:
So we got `backstage.io/managed-by-origin-location`, `backstage.io/managed-by-location`, and `backstage.io/source-location`, all referring to the source of the entity in different ways, at different levels? Perhaps a `location-ancestry` makes sense to bring it back down to just 2 annotations? With a list of all locations that were used to track down the entity.

Or am I missing the purpose of the 3 annotations?

Reply:
Yeah one for the actual registrered location (to keep track of which ones to unregister if you remove the location), one to do a single step of ancestry (can make several queries to stitch the full picture together), and one to point out the actual thing being described.

Reply:
Should have called them `registered-location`, `catalog-info-yaml-location` and `source-code-location` ;)

Reply:
So back to the original question, when is `source-code-location` not a URL?

Reply:
I guess we don't know yet. When the source is in a not-browser-based interface I guess. But isn't that the same as asking when will `backstage.io/managed-by-origin-location` not be a URL? 🤔 Maybe not

Reply:
So right now the [documentation](https://backstage.io/docs/features/software-catalog/well-known-annotations#backstageiosource-location) says that it's a 

> `Location` reference

so it would be correct to include `url:` and have the UI parse such location references. The UI would only work for `url:` type though. We might consider to add a small parser for the format to `catalog-model` as we already have multiple of them.

Reply:
Done, can you have another look?

Reply:
+1 been meaning to add those for quite some time, so i did :) https://github.com/backstage/backstage/pull/4835

Reply:
Yep, didn't mean to block this PR on the discussion, but it did need to be in sync with the schema

Reply:
I will wait for #4835 and remove the duplicate changes on monday, I guess than we are ready to go